pkgs/top-level/all-packages.nix

```diff
@@ -15036,6 +15044,13 @@ with pkgs;
     targetLlvm = targetPackages.llvmPackages_14.llvm or llvmPackages_14.llvm;
   }));
 
+  llvmPackages_15 = recurseIntoAttrs (callPackage ../development/compilers/llvm/15 ({
+    inherit (stdenvAdapters) overrideCC;
+    buildLlvmTools = buildPackages.llvmPackages_15.tools;
+    targetLlvmLibraries = targetPackages.llvmPackages_15.libraries or llvmPackages_15.libraries;
+    stdenv = llvmPackages_13.stdenv;
```

Comment:
Why does it need stdenv from llvm 13?

Reply:
Related: https://github.com/NixOS/nixpkgs/pull/184408

Reply:
They dropped support for clang12 and I tested on macos so stdenv is llvm)
Let me find the line with this info

Reply:
It probably should be under `isDarwin` condition since gcc is used to build llvm on Linux.

Reply:
https://github.com/llvm/llvm-project/blob/llvmorg-14.0.6/libcxx/include/__config#L423-L424

These line are removed on llvmorg-15.0.6, I didn't find requirement in README or in documentation so I don't know minimal version of gcc. You are right I will add a guard.

Reply:
I think It is not the correct way to require a recent version of clang because it has reference to llvm11 (from darwin stdenv):

```
$ nix-store -qR $(nix-store -qd /nix/store/n58zxqjx4lz785swd6xxcxv03la5lhfv-libcxxabi-15.0.6) | grep llvm
/nix/store/yww9hd80ikq72zrvkdk36picpqgvsh8b-llvm-11.1.0.src.tar.xz.drv
/nix/store/6wv8acdd7bm7npixgr3rk86mf55fj47c-llvm-config-link-static.patch
/nix/store/2mkz2cmqwg45xv7xfsj9idsb39maipmr-llvm-11.1.0.drv
/nix/store/cpnly3bmw6blsryxfgp2hs8ig2jz2aaz-llvm-13.0.1.drv
```